smartsim/entity/model.py

```diff
@@ -56,132 +48,151 @@
 # TODO: Remove this supression when we strip fileds/functionality
 #       (run-settings/batch_settings/params_as_args/etc)!
 # pylint: disable-next=too-many-public-methods
+
+
 class Application(SmartSimEntity):
     def __init__(
         self,
         name: str,
         exe: str,
-        run_settings: "RunSettings",
-        params: t.Optional[t.Dict[str, str]] = None,
-        exe_args: t.Optional[t.List[str]] = None,
-        params_as_args: t.Optional[t.List[str]] = None,
-        batch_settings: t.Optional["BatchSettings"] = None,
+        exe_args: t.Optional[t.Union[str, t.Sequence[str]]] = None,
         files: t.Optional[EntityFiles] = None,
-    ):
-        """Initialize a ``Application``
+        file_parameters: t.Mapping[str, str] | None = None,
+    ) -> None:
+        """Initialize an ``Application``
 
         :param name: name of the application
         :param exe: executable to run
         :param exe_args: executable arguments
-        :param params: application parameters for writing into configuration files or
-                       to be passed as command line arguments to executable.
-        :param run_settings: launcher settings specified in the experiment
-        :param params_as_args: list of parameters which have to be
-                               interpreted as command line arguments to
-                               be added to run_settings
-        :param batch_settings: Launcher settings for running the individual
-                               application as a batch job
-        :param files: Files to have available to the application
+        :param files: files to be copied, symlinked, and/or configured prior to
+                      execution
+        :param file_parameters: parameters and values to be used when configuring
+                                files
         """
-        super().__init__(name, run_settings)
-        self.exe = [expand_exe_path(exe)]
-        # self.exe = [exe] if run_settings.container else [expand_exe_path(exe)]
-        self.exe_args = exe_args or []
-        self.params = params.copy() if params else {}
-        self.params_as_args = params_as_args
-        self.incoming_entities: t.List[SmartSimEntity] = []
+        super().__init__(name)
+        self._exe = [expand_exe_path(exe)]
```

Comment:
Not a 'you' problem, but I'm kinda currious why `self._exe` is a `list[str]`. It might be worth bringing this to the group.

My guess is that it was at one point pulling double duty for `exe` and `exe_args`. I'm not sure that I like that and might request that we try to change it to a `str` in this PR.

Reply:
changed